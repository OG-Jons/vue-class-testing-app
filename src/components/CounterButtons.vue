<template>
  <button id="decrement" @click="count--">-</button>
  {{ count }}
  <button id="increment" @click="count++">+</button>
</template>

<script setup>
import { ref } from "vue";

const count = ref(0);
</script>

<style scoped lang="scss">
button {
  border: none;
  background: #42b983;
  padding: 0.5rem 2rem;
  margin: 0.5rem;

  &:active {
    background: #21704d;
  }
}
</style>
