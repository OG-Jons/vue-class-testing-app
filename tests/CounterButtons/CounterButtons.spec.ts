import { mount, VueWrapper } from "@vue/test-utils";
import CounterButtons from "../../src/components/CounterButtons.vue";

test("CounterButtons is mounted", async () => {
  const wrapper: VueWrapper = mount(CounterButtons);
  expect(CounterButtons).toBeTruthy();

  expect(wrapper.text()).toContain("0");
  expect(wrapper.html()).toMatchSnapshot();
});

test("increments", async () => {
  const wrapper: VueWrapper = mount(CounterButtons);

  await wrapper.get("#increment").trigger("click");

  expect(wrapper.text()).toContain("1");
});

test("decrements", async () => {
  const wrapper: VueWrapper = mount(CounterButtons);

  await wrapper.get("#decrement").trigger("click");

  expect(wrapper.text()).toContain("-1");
});
