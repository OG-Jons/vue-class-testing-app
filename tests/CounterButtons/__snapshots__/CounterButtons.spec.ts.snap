// Vitest Snapshot v1

exports[`CounterButtons is mounted 1`] = `
"<button id=\\"decrement\\" data-v-1036bcd6=\\"\\">-</button>
 0
<button id=\\"increment\\" data-v-1036bcd6=\\"\\">+</button>"
`;
